src/app/games/hive/tests/HiveRules.spec.ts

```diff
@@ -576,6 +576,20 @@ describe('HiveRules', () => {
             const reason: string = HiveFailure.GRASSHOPPER_MUST_MOVE_IN_STRAIGHT_LINE();
             RulesUtils.expectMoveFailure(rules, state, move, reason);
         });
+        it('should forbid moving the spider by a regular move instead of a spider move', () => {
+            // Given a board with a spider
+            const board: Table<HivePiece[]> = [
+                [[Q], [S], [q]],
+                [[A], [a], []],
+            ];
+            const state: HiveState = HiveState.fromRepresentation(board, 4);
+
+            // When moving the spider with a regular move
+            const move: HiveMove = HiveMove.move(new Coord(1, 0), new Coord(2, -1)).get();
+
+            // Then the move should fail
+            RulesUtils.expectMoveFailure(rules, state, move, HiveFailure.SPIDER_MUST_MOVE_WITH_SPIDER_MOVE);
```

Comment:
expect to throw and log in the db !

Reply:
en soit faudrait voir parfois ce qui est préférable entre assert et failure dans certains cas.
Si une IA est buggée et génère un mouvement illégal actuellement, ça throw une erreur dans les règles.
Si c'était une failure, c'est le vérificateur de move d'IA qui throwerait une erreur, ça pourrait être plus débuggable ?

Mais ici on est de nouveau sur un assert donc !

Reply:
Non mais c'est bien vu, faut que Rules.choose retourne un MGPValidation, pas un booleen, il pourrait transférer simplement la raison de l'erreur de move de l'IA ! Changement rapide à soit faire soit ticketter, ton choué !

Reply:
le booléen de Rules.choose il est jamais checké, donc je vois pas ce qu'on gagnerait à faire cela ? Ce qu'il se passe c'est que la légalité est vérifiée, puis ça fait du rules.choose. Je privilégie le tickettage car c'est clairement hors-scope et on est sur la fin du ticket ! J'sais pas trop quoi ticketter donc je te laisse faire si c'est toujours pertinent et puis résoudre !